backend/app/main.py

```diff
@@ -167,6 +169,12 @@
     tags=["groups"],
     dependencies=[Depends(get_current_username)],
 )
+api_router.include_router(
+    visualization.router,
+    prefix="/visualization",
```

Comment:
We have been using plurals for the custom routes. That's usually the case for rest.